pkg/configmap/configmap_watcher.go

```diff
@@ -31,54 +35,91 @@ type KubeLinterChecks struct {
-// dynamic: with StartInformer it sets an Informer that will be triggered on ConfigMap update
-func NewConfigMapWatcher(cfg *rest.Config) (ConfigMapWatcher, error) {
+// NewWatcher creates a new Watcher instance for observing changes to a ConfigMap.
+//
+// Parameters:
+//   - cfg: A pointer to a rest.Config representing the Kubernetes client configuration.
+//
+// Returns:
+//   - A Watcher instance for monitoring changes to DVO ConfigMap resource if the initialization is successful.
+//   - An error if there's an issue while initializing the Kubernetes clientset.
+func NewWatcher(cfg *rest.Config) (Watcher, error) {
 	clientset, err := kubernetes.NewForConfig(cfg)
 	if err != nil {
-		return ConfigMapWatcher{}, fmt.Errorf("initializing clientset: %w", err)
+		return Watcher{}, fmt.Errorf("initializing clientset: %w", err)
+	}
+
+	// the Informer will use this to monitor the namespace for the ConfigMap.
+	namespace, err := getDeploymentNamespace(clientset)
+	if err != nil {
+		return Watcher{}, fmt.Errorf("getting namespace: %w", err)
 	}
 
-	return ConfigMapWatcher{
+	// fmt.Printf("namespaces: %v\n", deployments)
+	return Watcher{
 		clientset: clientset,
 		logger:    log.Log.WithName("ConfigMapWatcher"),
+		ch:        make(chan config.Config),
+		namespace: namespace,
 	}, nil
 }
 
 // GetStaticKubelinterConfig returns the ConfigMap's checks configuration
-func (cmw *ConfigMapWatcher) GetStaticKubelinterConfig(ctx context.Context) (config.Config, error) {
+func (cmw *Watcher) GetStaticKubelinterConfig(ctx context.Context) (config.Config, error) {
 	cm, err := cmw.clientset.CoreV1().
-		ConfigMaps(configMapNamespace).Get(ctx, configMapName, v1.GetOptions{})
+		ConfigMaps(cmw.namespace).Get(ctx, configMapName, v1.GetOptions{})
 	if err != nil {
 		return config.Config{}, fmt.Errorf("getting initial configuration: %w", err)
 	}
 
 	return cmw.getKubeLinterConfig(cm.Data[configMapDataAccess])
 }
 
-// StartInformer will update the channel structure with new configuration data from ConfigMap update event
-func (cmw *ConfigMapWatcher) StartInformer(ctx context.Context) error {
+// Start will update the channel structure with new configuration data from ConfigMap update event
+func (cmw Watcher) Start(ctx context.Context) error {
 	factory := informers.NewSharedInformerFactoryWithOptions(
-		cmw.clientset, time.Second*30, informers.WithNamespace(configMapNamespace),
+		cmw.clientset, time.Second*30, informers.WithNamespace(cmw.namespace),
 	)
 	informer := factory.Core().V1().ConfigMaps().Informer()
 
 	informer.AddEventHandler(cache.ResourceEventHandlerFuncs{ // nolint:errcheck
+		AddFunc: func(obj interface{}) {
+			newCm := obj.(*apicorev1.ConfigMap)
+
+			if configMapName != newCm.GetName() {
+				return
+			}
+
+			cmw.logger.Info("ConfigMap has been created")
+
+			cfg, err := cmw.getKubeLinterConfig(newCm.Data[configMapDataAccess])
+			if err != nil {
+				cmw.logger.Error(err, "ConfigMap data format")
```

Comment:
similar here.... I am just wondeiing how this message would look like in the log :) integration tests will hopefully show us :)